.github/workflows/build-docker-image-and-binaries.yaml

```
@@ -36,58 +38,100 @@ jobs:
           submodules: recursive
 
       - name: Install Cosign
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         uses: sigstore/cosign-installer@main
 
-      - name: Import GPG key
-        id: import_gpg
-        uses: crazy-max/ghaction-import-gpg@v4
+      - name: Set up ssh agent
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
+        uses: webfactory/ssh-agent@v0.5.2
         with:
-          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
-          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+          ssh-private-key: ${{ secrets.CICD_RSA_KEY }}
 
       - name: Set up QEMU
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         uses: docker/setup-qemu-action@v1
 
       - name: Set up Docker Buildx
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         id: buildx
         uses: docker/setup-buildx-action@v1
         with:
           config: .github/buildkit-config.toml
 
       - name: Login to DockerHub
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         uses: docker/login-action@v1
         with:
           username: ${{ secrets.DOCKER_HUB_USERNAME }}
           password: ${{ secrets.DOCKER_HUB_TOKEN }}
 
       - name: Build and push docker image for all platforms
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         run: |
           make build-push-docker-images
         env:
           SEMVER: ${{ github.event.inputs.tag }}
 
       - name: Sign the images with GitHub OIDC
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         run: cosign sign --oidc-issuer https://token.actions.githubusercontent.com ${TAGS}
         env:
           TAGS: axelarnet/axelar-core:${{ github.event.inputs.tag }}
           COSIGN_EXPERIMENTAL: 1
 
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.5
+
       - name: Build Binaries for Linux/MacOS
         env:
           SEMVER: ${{ github.event.inputs.tag }}
         run: |
-          make build-binaries-in-docker
+          if [ "$RUNNER_OS" == "Linux" ]
+          then
+              OS="linux"
+          else
+              OS="darwin"
+          fi
+          ARCH="${{ matrix.arch }}"
+          make build
+          mv ./bin/axelard ./bin/axelard-"$OS"-"$ARCH"-"$SEMVER"
+
+      - name: Unit test (check ledger support)
+        run: |
+          ./bin/axelard-* version --long
```

```suggestion
          ./bin/axelard-"$OS"-"$ARCH"-"$SEMVER" version --long | grep "build_tags: ledger"
```
Don't you want to `grep` for `ledger` in the build_tags and check if it's present? And just to be precise